sympy/stats/drv_types.py

```diff
@@ -49,6 +50,64 @@ def rv(symbol, cls, *args):
     return pspace.value
 
 
+class DiscreteDistributionHandmade(SingleDiscreteDistribution):
+    _argnames = ('pdf',)
+
+    def __new__(cls, pdf, set=S.Integers):
+        cls.check(pdf, set)
+        return Basic.__new__(cls, pdf, set)
+
+    @property
+    def set(self):
+        return self.args[1]
+
+    @staticmethod
+    def check(pdf, set):
+        x = Dummy('x')
+        val = Sum(pdf(x), (x, set._inf, set._sup)).doit()
+        _value_check(val == S.One, "The pdf is incorrect on the given set.")
+
+def DiscreteRV(symbol, density, set=S.Integers):
+    """
+    Create a Discrete Random Variable given the following:
+
+    Parameters
+    ==========
+
+    symbol : Symbol
+        Represents name of the random variable.
+    density : Expression containing symbol
+        Represents probability density function.
+    set : set
+        Represents the region where the pdf is valid, by default is real line.
+
+    Examples
+    ========
+
+    >>> from sympy.stats import DiscreteRV
+    >>> from sympy import Rational, Symbol
+    >>> x = Symbol('x')
+    >>> n = 10
+    >>> density = Rational(1, 10)
+    >>> X = DiscreteRV(x, density, set=set(range(n)))
```

Comment:
A better way can be to pass `pmf` as a `dict`(or whatever the type used to store `pmf` inside `sympy.stats`).

Reply:
Actually, we use `pdf` for discrete distributions, with `pdf` as an expression. And `dict` is passed as `pmf` in Finite distributions.

Reply:
> And `dict` is passed as `pmf` in Finite distributions.

I see, can we pass `dict` as `pmf` to Finite distributions?

Reply:
Yes
https://github.com/sympy/sympy/blob/6daf556517f9fa61a7805fe93d8a366688781617/sympy/stats/frv_types.py#L74-L96